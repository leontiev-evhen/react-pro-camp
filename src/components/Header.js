import React from 'react';
import { NavLink } from 'react-router-dom';
import {
	Navbar,
	Nav,
	Form,
	FormControl,
	Button,
	Figure,
} from 'react-bootstrap';
import logo from '../assets/logo.png';

const NAV_ITEMS = [
	{
		link: '/',
		name: 'Home',
	},
	{
		link: '/teams',
		name: 'Teams',
	},
	{
		link: '/fixtures',
		name: 'Fixtures',
	},
	{
		link: '/odds',
		name: 'Odds',
	},
];

export function Header() {
	return (
		<header>
			<Navbar bg="light" expand="lg">
				<Navbar.Brand href="/">
					<Figure.Image
						src={logo}
						width="80"
						height="80"
						className="d-inline-block align-top"
						alt="React Bootstrap logo"
					/>
				</Navbar.Brand>
				<Navbar.Toggle aria-controls="basic-navbar-nav" />
				<Navbar.Collapse id="basic-navbar-nav">
					<Nav className="mr-auto">
						{NAV_ITEMS.map(item => (
							<NavLink
								key={item.link}
								to={item.link}
								className="nav-link"
								exact
								activeClassName="selected"
							>
								{item.name}
							</NavLink>
						))}
					</Nav>
					<Form inline>
						<FormControl type="text" placeholder="Search" className="mr-sm-2" />
						<Button variant="outline-secondary">Search</Button>
					</Form>
				</Navbar.Collapse>
			</Navbar>
		</header>
	);
}
