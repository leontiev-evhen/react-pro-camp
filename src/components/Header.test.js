import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
	const Component = shallow(<Header />);

	it('Header component render correctly', () => {
		expect(Component).toMatchSnapshot();
	});

	it('Header navigation links', () => {
		expect(Component.find('.nav-link')).toHaveLength(3);
	});
});
