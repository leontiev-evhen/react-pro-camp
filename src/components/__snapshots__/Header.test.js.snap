// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component render correctly 1`] = `
<header>
  <Bootstrap(undefined)
    bg="light"
    expand="lg"
  >
    <NavbarBrand
      href="/"
    >
      <FigureImage
        alt="React Bootstrap logo"
        className="d-inline-block align-top"
        fluid={true}
        height="80"
        src="logo.png"
        width="80"
      />
    </NavbarBrand>
    <NavbarToggle
      aria-controls="basic-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <NavLink
          activeClassName="selected"
          className="nav-link"
          exact={true}
          key="/"
          to="/"
        >
          Home
        </NavLink>
        <NavLink
          activeClassName="selected"
          className="nav-link"
          exact={true}
          key="/teams"
          to="/teams"
        >
          Teams
        </NavLink>
        <NavLink
          activeClassName="selected"
          className="nav-link"
          exact={true}
          key="/fixtures"
          to="/fixtures"
        >
          Fixtures
        </NavLink>
        <NavLink
          activeClassName="selected"
          className="nav-link"
          exact={true}
          key="/odds"
          to="/odds"
        >
          Odds
        </NavLink>
      </Nav>
      <Form
        inline={true}
      >
        <FormControl
          className="mr-sm-2"
          placeholder="Search"
          type="text"
        />
        <Button
          active={false}
          disabled={false}
          type="button"
          variant="outline-secondary"
        >
          Search
        </Button>
      </Form>
    </NavbarCollapse>
  </Bootstrap(undefined)>
</header>
`;
